File: subsequence.cpp
#include <string>

class Solution {
public:
  bool isSubsequence(std::string s, std::string t) {
    int sCount = 0;
    for (int i = 0; i < t.length(); i++) {
      if (t[i] == s[sCount]) {
        sCount++;
      }
    }

    if (sCount >= s.length()) {
      return true;
    }
    return false;
  }
};

File: subsequence.ts
function isSubsequence(s: string, t: string): boolean {
  let sCount = 0;

  for (let i = 0; i < t.length; i++) {
    if (t[i] === s[sCount]) {
      sCount++;
    }
  }

  if (sCount >= s.length) return true;
  return false;
}
